pkg/util/gziphandler/gzip.go

```diff
@@ -81,19 +86,12 @@ type GzipResponseWriter struct {
 
 	code int // Saves the WriteHeader value.
 
-	minSize int    // Specifed the minimum response size to gzip. If the response length is bigger than this value, it is compressed.
+	minSize int    // Specifies the minimum response size to gzip. If the response length is bigger than this value, it is compressed.
 	buf     []byte // Holds the first part of the write before reaching the minSize or the end of the write.
 	ignore  bool   // If true, then we immediately passthru writes to the underlying ResponseWriter.
 
-	contentTypes []parsedContentType // Only compress if the response is one of these content-types. All are accepted if empty.
-}
-
-type GzipResponseWriterWithCloseNotify struct {
-	*GzipResponseWriter
-}
-
-func (w GzipResponseWriterWithCloseNotify) CloseNotify() <-chan bool {
-	return w.ResponseWriter.(http.CloseNotifier).CloseNotify()
+	contentTypes    []parsedContentType // Only compress if the response is one of these content-types. All are accepted if empty.
+	acceptsIdentity bool                // If false, then request explicitly rejected non-encoded requests.
```

Comment:
The comment here seems like a double negative and I'm confused. If `acceptsIdentity` is false, then we reject non-encoded requests? The extra phrasing in the comment is throwing me off

Reply:
Hopefully it's better now since https://github.com/grafana/mimir/pull/1864/commits/705540dcb40c16de631b778770a53ccc6b079291 as part of the refactor I mentioned below.